app/services/anno_auth_validator.rb

```diff
@@ -0,0 +1,43 @@
+require 'json'
+require 'net/http'
+require 'jwt'
+
+class AnnoAuthValidator
+
+    def initialize(authHeader, canvas)
+        @authHeader = authHeader
+        @canvas = canvas
+      end
+
+    # Verfiying from remote destination that the token sent has the same group ID for the canvas and that the user has access
+    def authorize
+        begin
+        token = decodeToken(@authHeader)
```

Comment:
I'd personally add private getters to your instance variables, i.e. `attr_reader :auth_header, :canvas` after the `private` keyword. This is mostly stylistic.